src/app/pages/animais/animais.component.ts

```diff
@@ -21,16 +27,39 @@ export class AnimaisComponent implements OnInit {
     this.router.navigate([`/private/animal`]);
   }
 
-  excluir(id: number) {
-    this.animaisService.deletar(id);
+  async excluir(id: number): Promise<void> {
+    const res = await this.animaisService.deletar(id);
+
+    if (res) {
+      this.toast.mostrarSucesso('Excluído com sucesso!');
+    } else {
+      this.toast.mostrarErro(
+        'O Animal não pode ser excluido. Já  foi feito atendimento nesse fidido.'
+      );
+    }
   }
 
   editar(id: number) {
     this.router.navigate([`/private/animal/${id}`]);
   }
 
   async pegarDadosTabela(): Promise<void> {
-    const input = await this.animaisService.buscarTodos();
-    this.listarAnimais = input;
+    const res = await this.animaisService.buscarTodos();
+    if (res) {
+      const input = res.filter((el) => el.ativo === true);
+      this.listarAnimais = input;
+    }
+  }
+
+  async filtrar(): Promise<void> {
+    const res = await this.animaisService.buscarTodos();
+    if (res) {
+      if (this.mostrarAtivos) {
```

Comment:
Se o this.mostrarAtivos retorna um booleano vc não precisa fazer essa condição e usar duas vezes o filter.